satellite/repair/repairer/ec.go

```diff
@@ -115,9 +115,12 @@ func (ec *ECRepairer) Get(ctx context.Context, limits []*pb.AddressedOrderLimit,
 				cond.L.Lock()
 				inProgress--
 				if err != nil {
-					// gather nodes that failed to match piece hash with the original piece hash
+					// gather nodes where the calculated piece hash doesn't match the uplink signed piece hash
 					if ErrPieceHashVerifyFailed.Has(err) {
-						failedNodes = append(failedNodes, limit.GetLimit().StorageNodeId)
+						failedPieces = append(failedPieces, &pb.RemotePiece{
+							PieceNum: int32(currentLimitIndex),
+							NodeId:   limit.GetLimit().StorageNodeId,
+						})
```

Comment:
`currentLimitIndex` seems to be the array index.  does the array index always equal the `PieceNum`?

Reply:
yeah, the limits are generated with `orders.CreateGetRepairOrderLimits`, which uses the piece number to index the limits.